lib/phoenix/controller/flash.ex

```diff
@@ -0,0 +1,97 @@
+defmodule Phoenix.Controller.Flash do
```

Comment:
Mostly now verbatim from `Phoenix.LiveView.Flash`.

My intention however, is that all flash capabilities move from the controller into this helper and it incorporates the session retrieval of the flash message.  The `fetch_flash` and other related methods will be then `defdelegate`s into here.

Reply:
Thanks @aforward!

What if we start in the opposite direction: first we create `Phoenix.Controller.Flash` without any of the liveview code and then we add the liveview code to it?

Also, regarding the signing salt, it would be nice if we could use the session signing salt. But we need to check if it is exposed in any way. I will investigate.

Reply:
hehe, ok.

that's how I had structured the original commits.

Reply:
Oh, i forgot to check the commits individually! Sorry for the confusion then!

Reply:
@josevalim I have reverted to the original PR, but rolled back the last commit that incorporated LiveView functionality.

And not having a separate salt would be great, let me know.

Reply:
Sidebar: I really ❤️ git, these back and forth have been a breeze with `git rebase -i HEAD~X`... I wanted to keep this PR conversation so the only unfortunate part is the branch name having to stay unchanged.